### PR TITLE
DEPR: Remove legacy offsets

### PR DESCRIPTION
**Pull request by sinhrks**
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

Follow-up of #10951. Remove legacy offsets deprecated in 0.17.0.

### CODE REVIEW

File: doc/source/whatsnew/v0.19.0.txt [hunk 1]
```diff
@@ -432,6 +432,7 @@ Deprecations
 - ``buffer_lines`` has been deprecated in ``pd.read_csv()`` and will be removed in a future version (:issue:`13360`)
 - ``as_recarray`` has been deprecated in ``pd.read_csv()`` and will be removed in a future version (:issue:`13373`)
 - top-level ``pd.ordered_merge()`` has been renamed to ``pd.merge_ordered()`` and the original name will be removed in a future version (:issue:`13358`)
+- Removal of the legacy time rules (offset aliases), deprecated since 0.14.0 (this has been alias since 0.8.0) (GH13590).
```
Comment (jreback) [inline doc/source/whatsnew/v0.19.0.txt]: put in a top-level separate section (prior deprecation removal - look how I did this for 0.18.0)

File: doc/source/whatsnew/v0.19.0.txt [hunk 2]
```diff
@@ -433,6 +433,13 @@ Deprecations
 - ``as_recarray`` has been deprecated in ``pd.read_csv()`` and will be removed in a future version (:issue:`13373`)
 - top-level ``pd.ordered_merge()`` has been renamed to ``pd.merge_ordered()`` and the original name will be removed in a future version (:issue:`13358`)
 
+.. _whatsnew_0190.prior_deprecations:
+
+Removal of prior version deprecations/changes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Removal of the legacy time rules (offset aliases), deprecated since 0.14.0 (this has been alias since 0.8.0) (GH13590).
+
```
Comment (jreback) [inline doc/source/whatsnew/v0.19.0.txt]: put the issue reference

Comment (jreback) [inline doc/source/whatsnew/v0.19.0.txt]: I would also add a blurb about what exactly these are (show an example), and point to the new docs as well.

**Comment by jreback** [comment]
lgtm otherwise.

File: doc/source/whatsnew/v0.19.0.txt (hunk 2 above)
Comment (sinhrks) [inline doc/source/whatsnew/v0.19.0.txt]: oops, changed the issue number to the link.

**Comment by jorisvandenbossche** [comment]
The "Removal of prior version deprecations/changes" section has already been added in the meantime, so I think this needs a rebase

**Pull request closed without merging**